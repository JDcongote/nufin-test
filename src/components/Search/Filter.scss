@import '../../styles/variables/index';
@import '../../styles/tools/mixins';

.filter {
  display: flex;
  justify-content: center;

  @include breakpoint(tablet-portrait-up) {
    width: 100%;
  }

  &__select {
    -webkit-appearance: none;
    background-color: var(--color--secondary);
    border: none;
    width: 100%;
    cursor: pointer;
    outline: none;
    text-transform: capitalize;
    opacity: 0.5;

    &-container {
      margin: 10px 1em;
      border-radius: var(--border--radius--normal);
      border: none;
      padding: 0.5em 1em;
      width: 100%;
      background-color: var(--color--secondary);
      display: flex;

      &:after {
        content: '';
        width: 5px;
        height: 5px;
        border-bottom: 2px solid var(--color--primary);
        border-right: 2px solid var(--color--primary);
        transform: rotateZ(45deg);
        align-self: center;
        margin-left: auto;
      }
    }
  }
}
