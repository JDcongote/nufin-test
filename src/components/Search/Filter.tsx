import React from 'react';
import './Filter.scss';

type FilterProps = {
  onFilter: (filtered: any[]) => void;
  filters: tFilter[];
  content: any[];
  property: string;
};

export type tFilter = {
  name: string;
  id: string;
};

type State = {};

class Filter extends React.PureComponent<FilterProps, State> {
  componentDidMount() {
    this.setState({ searchTerm: '' });
  }
  doFiltering(event: React.BaseSyntheticEvent) {
    const term = event.target.value.toLowerCase();
    if (term !== '-') {
      const filtered = this.props.content.filter(
        item => item[this.props.property].toLowerCase().indexOf(term) > -1
      );
      this.props.onFilter(filtered);
    } else {
      this.props.onFilter(this.props.content);
    }
  }
  render() {
    return (
      <div className="filter">
        <div className="filter__select-container">
          <select
            className="filter__select"
            onChange={this.doFiltering.bind(this)}
          >
            <option
              key="default"
              id="none"
              onClick={this.doFiltering.bind(this)}
            >
              -
            </option>
            {this.props.filters.map(filter => (
              <option key={filter.id} id={filter.id}>
                {filter.name}
              </option>
            ))}
          </select>
        </div>
      </div>
    );
  }
}

export default Filter;
