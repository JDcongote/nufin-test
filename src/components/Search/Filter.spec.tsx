import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import renderer from 'react-test-renderer';
import Filter, { tFilter } from './Filter';

configure({ adapter: new Adapter() });

describe('Filter', () => {
  it('Basic Filter Test', () => {
    const content = [{ id: '1', name: 'a' }, { id: '2', name: 'b' }];
    const onFilter = jest.fn();
    const property = 'name';
    const filters: tFilter[] = [{ name: 'a', id: '1' }, { name: 'a', id: '2' }];

    const render = renderer.create(
      <Filter
        content={content}
        onFilter={onFilter}
        property={property}
        filters={filters}
      />
    );
    expect(render.toJSON()).toMatchSnapshot();
  });

  it('Search test', () => {
    const content = [{ id: '1', name: 'a' }, { id: '2', name: 'b' }];
    const onFilter = val => {
      expect(val.length).toBeGreaterThan(0);
      expect(val.name).toBe('a');
    };
    const property = 'name';
    const filters: tFilter[] = [{ name: 'a', id: '1' }, { name: 'a', id: '2' }];

    const wrapper = mount(
      <Filter
        content={content}
        onFilter={onFilter}
        property={property}
        filters={filters}
      />
    );
    wrapper
      .find('.filter__select')
      .find('option')
      .at(1)
      .simulate('click');
  });
});
