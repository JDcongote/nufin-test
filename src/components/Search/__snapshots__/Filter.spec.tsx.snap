// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter Basic Filter Test 1`] = `
<div
  className="filter"
>
  <div
    className="filter__select-container"
  >
    <select
      className="filter__select"
      onChange={[Function]}
    >
      <option
        id="none"
        onClick={[Function]}
      >
        name
      </option>
      <option
        id="1"
      >
        a
      </option>
      <option
        id="2"
      >
        a
      </option>
    </select>
  </div>
</div>
`;
